app/controllers/share_controller.rb

```diff
@@ -0,0 +1,26 @@
+class ShareController < ApplicationController
+
+  before_action :find_post
+
+  def create
+    flash[:success] = "shared!"
+    UserMailer.share(@post, share_params).deliver
```

Comment:
Does this work? Offhand, I'd expect `share_params` to be a hash like: `{:email => 'email@example.org'}`